adapter/pkg/operator/utils/utils.go

```diff
@@ -269,3 +276,107 @@ func GetSecretValue(ctx context.Context, client client.Client,
 	}
 	return string(secret.Data[key]), nil
 }
+
+// ResolveAndAddBackendToMapping resolves backend from reference and adds it to the backendMapping.
+func ResolveAndAddBackendToMapping(ctx context.Context, client k8client.Client,
+	backendMapping dpv1alpha1.BackendMapping,
+	backendRef dpv1alpha1.BackendReference, apiPolicyNamespace string) {
+	namespace := gwapiv1b1.Namespace(backendRef.Namespace)
+	backendName := types.NamespacedName{
+		Name:      backendRef.Name,
+		Namespace: GetNamespace(&namespace, apiPolicyNamespace),
+	}
+	backend := GetResolvedBackend(ctx, client, backendName)
+	backendMapping[backendName] = backend
+}
+
+// GetResolvedBackend resolves backend TLS configurations.
+func GetResolvedBackend(ctx context.Context, client k8client.Client,
+	backendNamespacedName types.NamespacedName) *dpv1alpha1.ResolvedBackend {
```

Comment:
Isn't it better to return err and handle/log errors there? because these errors might be temperory resolving issues and better to log as warnings then
```suggestion
	backendNamespacedName types.NamespacedName) (*dpv1alpha1.ResolvedBackend, err) {
```

Reply:
error is actually logged  inside the function in line number 303 - `loggers.LoggerAPKOperator.ErrorC(logging.GetErrorByCode(2637, backendNamespacedName, err.Error()))`